packages/ui-components/src/Badge/Badge.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { FunctionComponent } from "react";
+import classNames from "classnames/bind";
+
+import styles from "./Badge.module.scss";
+
+type BadgeCase = "none" | "uppercase";
+type BadgeIntent = "neutral" | "success" | "warning" | "danger";
+
+export interface BadgeProps {
+  content?: string;
+  intent?: BadgeIntent;
+  transformCase?: BadgeCase;
+}
+
+const cx = classNames.bind(styles);
+
+const Badge: FunctionComponent<BadgeProps> = ({
+  content,
+  intent = "neutral",
+  transformCase = "uppercase",
+  children,
+  ...props
+}) => {
+  const classnames = cx("badge", {
+    "intent--success": intent === "success",
+    "intent--danger": intent === "danger",
+    "intent--warning": intent === "warning",
+    "intent--neutral": intent === "neutral",
+    "transformCase--uppercase": transformCase === "uppercase",
+    "transformCase--none": transformCase === "none",
```

Comment:
@nathanstilwell , as a shortcut for assigning classes depending on props might be useful in case new `intent` or `transformCase` options will be added:
```javascript
const classnames = cx("badge", `intent--${intent}`, `transformCase--${transformCase}`)
```
Having default values, it is safe to assume that only valid values will be provided

Reply:
I think this is a valid solution to tighten up this code. I'm proposing a utility to solve this [here](https://github.com/cockroachdb/ui/pull/4) and plan to refactor using that. I just left the _exploded_ version here to be explicit and perhaps illustrative.